server/events.go

```diff
@@ -404,6 +420,7 @@ RESET:
 					c.flushClients(time.Second)
 					return
 				}
+				pubMsgPool.Put(pmi)
```

Comment:
If there are interior items that are on the heap and could affect the GC I might set those to nil.
I do this in the current PR I have up as well.

Reply:
I was thinking about this and may then do a pubMsg.done() that will reset some (all?) fields before putting back to the pull.

Reply:
Only the ones that are externally stored. Yes.

Reply:
Done. Did it for strings too, not sure if required...

Reply:
Yes on strings IMO, so that is good.